### PR TITLE
Apply cal match freqs

### PR DESCRIPTION
**Pull request by anguta**
Allow for applying calibration files to data with intersecting but not identical frequency channels.

### CODE REVIEW

File: hera_cal/apply_cal.py [hunk 1]
```diff
@@ -246,11 +246,28 @@ def apply_cal(data_infilename, data_outfilename, new_calibration, old_calibratio
     # load old calibration solution
     if old_calibration is not None:
         old_hc = io.HERACal(old_calibration)
-        old_gains, old_flags, _, _ = old_hc.read()
+        old_hc.read()
+        # determine frequencies to load in old_hc that are close to hc
+        freqs_to_load = []
+        for f in old_hc.freqs:
+            atol = np.mean(np.diff(hc.freqs)) / 10.
```
Comment (jsdillon) [inline hera_cal/apply_cal.py:253]: It's not good to have magic numbers like this in the code. This tolerance factor, 10, should be a kwarg, though I think 10 is a reasonable default.

File: hera_cal/io.py [hunk 2]
```diff
@@ -641,7 +641,7 @@ def partial_write(self, output_path, data=None, flags=None, nsamples=None,
                                   this.nsample_array, **self.last_read_kwargs)
 
     def iterate_over_bls(self, Nbls=1, bls=None, chunk_by_redundant_group=False, reds=None,
-                         bl_error_tol=1.0, include_autos=True):
+                         bl_error_tol=1.0, include_autos=True, freqs_to_load=None):
```
Comment (jsdillon) [inline hera_cal/io.py:644]: To better match other iterators, I'd call this `freqs` or `frequencies` rather than `freqs_to_load`

File: hera_cal/io.py [hunk 3]
```diff
@@ -713,7 +716,10 @@ def iterate_over_bls(self, Nbls=1, bls=None, chunk_by_redundant_group=False, red
             # make sure that every baseline is in reds
             baseline_chunks = chunk_baselines_by_redundant_groups(reds=reds, max_chunk_size=Nbls)
         for chunk in baseline_chunks:
-            yield self.read(bls=chunk)
+            if freqs_to_load is None:
```
Comment (jsdillon) [inline hera_cal/io.py:719]: Do you actually need this if / else? If freqs is None, won't self.read behave appropriately?

File: scripts/apply_cal.py [hunk 4]
```diff
@@ -17,6 +17,9 @@
 if args.vis_units is not None:
     kwargs['vis_units'] = args.vis_units
 
+if args.nbl_per_load == 0:
```
Comment (jsdillon) [inline scripts/apply_cal.py:20]: Why is this necessary? When would this ever happen? Shouldn't this be an error?

**Review by jsdillon — CHANGES_REQUESTED** [review]
Looks pretty good. Just a few small comments/questions.

File: hera_cal/apply_cal.py (hunk 1 above)
Comment (anguta) [reply hera_cal/apply_cal.py:253]: i've added this to the kwargs and set default to 10.

File: hera_cal/io.py (hunk 3 above)
Comment (anguta) [reply hera_cal/io.py:719]: you're right, if freqs i None, things will work fine.

File: hera_cal/io.py (hunk 2 above)
Comment (anguta) [reply hera_cal/io.py:644]: calling it frequencies.

File: scripts/apply_cal.py (hunk 4 above)
Comment (anguta) [reply scripts/apply_cal.py:20]: this is to allow for some flexibility in the makeflow bash interface so that we can use the same scripts for when we use partial i/o vs no partial i/o. 

0 is confusing though. I've changed this to "none".

File: hera_cal/io.py [hunk 5]
```diff
@@ -667,7 +667,7 @@ def iterate_over_bls(self, Nbls=1, bls=None, chunk_by_redundant_group=False, red
             include_autos: bool, optional
                 include autocorrelations in iteration if True.
                 Default is True.
-            freqs_to_load: array-like, optional
+            freqsuencies: array-like, optional
```
Comment (jsdillon) [inline hera_cal/io.py:670]: typo

**Review by jsdillon — APPROVED** [review]
Just one typo, then good to go.

File: hera_cal/io.py (hunk 5 above)
Comment (anguta) [reply hera_cal/io.py]: fixed.
